### PR TITLE
Fix problem with incorrect mappings created for content indices

### PR DESCRIPTION
**Pull request by artem-shelkovnikov**
## Closes https://github.com/elastic/enterprise-search-team/issues/4465

When framework attempts to create a search index mappings, it does not actually provide "dynamic" and "dynamic_templates" settings to the `put_mappings` call. This PR changes it and actually starts providing these options.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

**Review by timgrein — APPROVED** [review]
LGTM

File: connectors/tests/test_byoei.py [hunk 1]
```diff
@@ -84,7 +85,17 @@ async def test_prepare_content_index(mock_responses):
...
+    dynamic_templates = {
+        "data": {
+            "match_mapping_type": "string",
+            "mapping": TEXT_FIELD_MAPPING,
+        }
+    }
+    mappings = {
+        "dynamic": True,
+        "dynamic_templates": dynamic_templates,
+        "properties": {"name": {"type": "keyword"}},
```
Comment (seanstory) [inline connectors/tests/test_byoei.py:97]: I see that this was in here before... but I don't think it should be? A `name` property is not in the ruby connectors' mappings, it has `id`.

https://github.com/elastic/connectors-ruby/blob/main/lib/utility/elasticsearch/index/mappings.rb#L92

```suggestion
        "properties": {"id": {"type": "keyword"}},
```

Comment (artem-shelkovnikov) [reply connectors/tests/test_byoei.py:97]: It should be fine as this specific function allows arbitrary mappings and this test verifies that these arbitrary mappings are passed correctly.

If we wanna test that our default mappings are passed properly, it should be in a different place, the one that calls `prepare_content_index`

**Comment by artem-shelkovnikov** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.8|[](https://github.com/elastic/connectors-python/pull/913)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**